dts/bindings/stepper_motor/stepper-motor-device.yaml

```diff
@@ -0,0 +1,58 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+# SPDX-License-Identifier: Apache-2.0
+
+description: Stepper Motor Device
+
+include: base.yaml
+
+on-bus: motor
+
+compatible: "stepper-motor-device"
+
+properties:
+
+  friendly-name:
+    type: string
+    description: |
+      Human readable string describing the stepper motor. It can be used to
+      distinguish multiple instances of the same model.
+
+  gear-ratio:
+    type: string
+    default: "1.0"
+    description: |
+      Gear ratio from stepper motor to end transmission, This value can be in float and hence
+      the property type string
+
+  micro-step-res:
+    type: int
+    enum:
+      - 1
+      - 2
+      - 4
+      - 8
+      - 16
+      - 32
+      - 64
+      - 128
+      - 256
+    description: |
+      Enter the microstep resolution from one of the enum values
```

Comment:
isn't this specialized to tmc5041?

Reply:
Hmm not really, as far as i know microsteps are always in the power of 2. Here an excerpt from DRV8825
<img width="721" alt="MicrosoftTeams-image" src="https://github.com/zephyrproject-rtos/zephyr/assets/126553325/9b68f19e-d8d1-4ab8-9bf2-22d766c2f6eb">